specifics/Dhyana/DhyanaStateMachine.cpp

```diff
@@ -63,22 +63,20 @@ namespace Dhyana_ns
 bool Dhyana::is_tucamVersion_allowed(TANGO_UNUSED(Tango::AttReqType type))
 {
 
-	if (get_state() == Tango::INIT ||
-		get_state() == Tango::FAULT ||
-		get_state() == Tango::RUNNING)
+	//	Check access type.
+	if ( type==Tango::READ_REQ )
 	{
-		//	End of Generated Code
-		if (get_state() == Tango::RUNNING && type == Tango::READ_REQ)
+		//	Compare device state with not allowed states for READ 
+		if (get_state()==Tango::INIT ||
+			get_state()==Tango::FAULT ||
+			get_state()==Tango::RUNNING)
 		{
-			return true;
+		/*----- PROTECTED REGION ID(Dhyana::tucamVersionStateAllowed_READ) ENABLED START -----*/
+		
+		/*----- PROTECTED REGION END -----*/	//	Dhyana::tucamVersionStateAllowed_READ
+			return false;
 		}
-
-		if (get_state() == Tango::FAULT && is_device_initialized())

```

Comment:
the function 'is_device_initialized()' is no more used: is it ok ? if yes, you should also remove it from the .h